### DISCUSSION TITLE
Stuck at Pad file check.

### DISCUSSION THREAD

**Discussion by cacafuti8888** [xCuri0/ReBarUEFI#158] [General]
**System**
 - Motherboard: ASUS MAXIMUS VIII RANGER
 - BIOS Version: 3802 (Modded to enable TPM)
 - GPU: XFX AMD RADEON RX580 
 - [*] CSM is turned off. **Make sure to confirm this in the BIOS and not with GPU-Z or similar since it can be inaccurate**
 - [*] 4G decoding is enabled. **Make sure to confirm this in the BIOS and not with GPU-Z or similar since it can be inaccurate**
 - [*] UEFIPatch is applied (see [Using UEFIPatch](https://github.com/xCuri0/ReBarUEFI/wiki/Using-UEFIPatch) for more information). On some motherboards [DSDT Patching](https://github.com/xCuri0/ReBarUEFI/wiki/DSDT-Patching) is also needed
 - [*] I have read [Common issues (and fixes)](https://github.com/xCuri0/ReBarUEFI/wiki/Common-issues-(and-fixes))

**Description**

I need assistance with some steps of the process. After succesfully run UEFIPatch i need to check if the pad section of the BIOS has changed but i don't know wich of the pad sections i have to look at, as i notice several pad sections in different parts of the BIOS file, one of wich its outside the patched zone and that has differences between the unpatched and the patched one.

I tried to follow your steps into adding this pad zone using MMTool but when i want to save it  it gave me an error.

I have all the programs requiered with matching versions so i don't know what i'm missing

**Comment by xCuri0** [comment]
try using MMTool 5

**Reply by cacafuti8888** [reply]
I tried both, UEFITool and MMTool to add the rebarmod ffs without issues. Problem is after UEFIPatch, when you asked to check if the pad files are misaligned or moved. Which ones? I have several pad in my BIOS. I detected one that it's removed afer i did UEFIPatch step. Is this ok?

![REBARMOD](https://github.com/xCuri0/ReBarUEFI/assets/16649341/b71b28ed-9749-4041-b605-daf68b021347)

**Comment by cacafuti8888** [comment]
Finally having a CH341a programmer and a couple of spare bios chips i flashed it and it boot no problems. And after applying the registry patch for unsupported GPU i applied 32 on ReBarState without any issue. Now I have rebar enabled with maximum memory shared. AMAZING. Thanks for the help, and for creating such an amazing tutorial and tools. Here are some screenshots

![Adrenalin ReBar on RX580](https://github.com/xCuri0/ReBarUEFI/assets/16649341/3432b062-f8c9-42bf-b5c7-6b4b4cd755c0)
![GPU-Z Rebar](https://github.com/xCuri0/ReBarUEFI/assets/16649341/e158ae57-c66e-4020-8538-dd1511013df4)
![Device Manager](https://github.com/xCuri0/ReBarUEFI/assets/16649341/c9ff2085-9959-4c08-bfcc-34dc5f15f9e0)

**Comment by SlowerBeen** [comment]
What trick have you done to solve this issue ? i dont any answers from no one, refering to your pad files problem

**Reply by cacafuti8888** [reply]
As I mentioned, I have an eeprom programmer. I flashed that bios and if it failed to boot I can flash a backup using other pc no problem.

If you don't have one, I don't recommend trying if you are not sure. As far I can tell, for me, it work first time no problem.

**Reply by SlowerBeen** [reply]
i have my biosflash back cover.
i brick my bios 5 times from when i started working on this mod, so luckly i dont ned an external programmer.

So my question is .. you are saying that your bios worked even whit padding issue ?

**Reply by cacafuti8888** [reply]
> So my question is .. you are saying that your bios worked even whit padding issue ? 

Well at least the padding that I showed in the picture above. 

I'm going to tell the steps that I followed and you can decide from there. 

First I have a prior modded bios to enable TPM on a non "compatible" (according to microsoft) windows 11 cpu. It's not a major difference from stock bios. 

So, I dumped my bios to a file and started the process to the rebarmod using UEFITool and following the guide. Next I used UEFIPatch which threw me 2 changes. It's with UEFIPatch when the padding occurs and the reason of my doubts to create this discussion. 

He answered me to use mmtool so I started the whole process using mmtool method. Then UEFIPatch, 2 changes again. Same padding than before. So it's definitely UEFIPatch who made those changes. It's there where I didn't know what to do but I realized that why not try and see if it works, I have bios Backups and a ch341a. And it worked first time. Seems that my padding was not a problem after all, but It may not the case  for your motherboard brand and model.

I wish to help but I can't offer much more.

**Reply by SlowerBeen** [reply]
Thank you man, your experience is important. :)

So, know i will think and decide what to do.
